pkg/influx/api.go

```diff
@@ -25,47 +24,23 @@ var ingesterClientRequestDuration = promauto.NewHistogramVec(prometheus.Histogra
 	Buckets:   prometheus.ExponentialBuckets(0.001, 4, 6),
 }, []string{"operation", "status_code"})
 
-type APIConfig struct {
-	DistributorEndpoint     string
-	DistributorClientConfig grpcclient.Config
-}
-
-func (c *APIConfig) RegisterFlags(flags *flag.FlagSet) {
-	flags.StringVar(&c.DistributorEndpoint, "distributor.endpoint", "", "The grpc endpoint for downstream Cortex distributor.")
-
-	c.DistributorClientConfig.RegisterFlagsWithPrefix("distributor.client", flags)
-}
-
 type API struct {
 	logger   log.Logger
-	client   distributorpb.DistributorClient
+	client   remotewrite.Client
 	recorder Recorder
 }
 
 func (a *API) Register(server *server.Server, authMiddleware middleware.Interface) {
 	server.HTTP.Handle("/api/v1/push/influx/write", authMiddleware.Wrap(http.HandlerFunc(a.handleSeriesPush)))
 }
 
-func NewAPI(logger log.Logger, cfg APIConfig) (*API, error) {
-	dialOpts, err := cfg.DistributorClientConfig.DialOption(grpcclient.Instrument(ingesterClientRequestDuration))
-	if err != nil {
-		return nil, err
-	}
-
-	level.Info(logger).Log("msg", "Creating GRPC connection to", "addr", cfg.DistributorEndpoint)
-	conn, err := grpc.Dial(cfg.DistributorEndpoint, dialOpts...)
-	if err != nil {
-		level.Error(logger).Log("msg", "Failed to connect to server", "err", err)
-		return nil, err
-	}
-
-	distClient := distributorpb.NewDistributorClient(conn)
+func NewAPI(logger log.Logger, client remotewrite.Client) (*API, error) {
 
 	recorder := NewRecorder(prometheus.NewRegistry())
```

Comment:
Just saw this but instead of saying prometheus.NewRegistry() you should pass it to the API from the caller or pass the recorder initialized.

In the main file you should have a registry variable i.e `reg` initialized to the registry you want to use for the entire program. This is useful later if you do tests because you can have test using different registries and test which metrics are registered for a specific usecase and so on.

Maybe its out of the scope of this PR but just bear this in mind :+1: